### PR TITLE
Control nodes

### PR DESCRIPTION
**Pull request by ollmer**
- rewrote gaia, workarena and math agents to the linear node flows
- introduced the control flow node and fixed steps node
- removed monoagents
- removed dated GaiaResults storage format
- updated tests with tapes containing SetNextNode step
- removed `.with_...()` methods from the Agent

### CODE REVIEW

File: tapeagents/agent.py [hunk 1]
```diff
@@ -74,51 +74,21 @@ class Node(BaseModel):
...
-        self.make_llm_output_func = make_llm_output
-        return self
-
-    def with_fixed_steps(self, steps: list[Step]) -> Node:
-        """
-        Use fixed steps instead of generating them from the LLM output for that node
-        """
-        self.generate_steps_func = lambda agent, tape, llm_stream: (yield from steps)  # type: ignore
-        return self
+        return LLMOutput(role="assistant", content=tape.steps[index].content)
```
Comment (rizar) [inline tapeagents/agent.py:91]: good riddance!

**Review by rizar — CHANGES_REQUESTED** [review]
thanks! I'd like to change two things:

- GuidanceNode -> MonoNode
- let's use absolute node indices to prevent confusion

File: examples/gaia_agent/agent.py [hunk 2]
```diff
@@ -114,74 +118,55 @@ def prepare_guidance(cls, planning_mode: PlanningMode, subtasks: bool) -> list[G
...
-                    guidance=PromptRegistry.think_after_observation,
-                ),
-                GaiaNode(
-                    name="think_after_calculation",
-                    trigger_step=["calculation_result_observation"],
-                    guidance=PromptRegistry.think_after_calculation,
+                GaiaNode(name="plan", guidance=PromptRegistry.plan),
+                GaiaNode(name="facts_survey", guidance=PromptRegistry.facts_survey),
+                GaiaNode(name="start_execution", guidance=PromptRegistry.start_execution, next_node=-1),
+                GaiaNode(name="think_after_observation", guidance=PromptRegistry.think_after_observation, next_node=-1),
```
[the remark this answers is not in the document]
Comment (ollmer) [reply examples/gaia_agent/agent.py:131]: Ok!

**Comment by ollmer** [comment]
@rizar all comments addressed, pleas take a look

File: tapeagents/nodes.py [hunk 3]
```diff
@@ -23,17 +25,17 @@
 
 class MonoNode(Node):
     """
-    A node for the monolithic agent.
+    A node for simple agents:
```
Comment (rizar) [inline tapeagents/nodes.py:28]:
```suggestion
    A node for simple monolithic agents:
```

**Pull request merged**